packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualPositionManager.sol

```diff
@@ -564,7 +563,7 @@ contract PerpetualPositionManager is FeePayer, FundingRateApplier, Administratee
      * @dev This will drain down to the amount of tracked collateral and drain the full balance of any other token.
      * @param token address of the ERC20 token whose excess balance should be drained.
      */
-    function trimExcess(IERC20 token) external fees() nonReentrant() returns (FixedPoint.Unsigned memory amount) {
+    function trimExcess(IERC20 token) external nonReentrant() returns (FixedPoint.Unsigned memory amount) {
```

Comment:
I'm guessing fees() was removed to save bytecode, but couldn't this cause a situation where the excess collateral that is trimmed eats into the amount that is supposed to be charged as regular fees?

Reply:
I don't think so (but check my logic :)). This only trims collateral that is not tracked by pfc(). The fees only consider the collateral that is tracked by pfc() (and the fees are removed from pfc()). So I think it shouldn't affect the amount that's charged on the next call to pfc().

Reply:
In theory I see your point. provided trim excess does not touch the PFC then this makes sense. We could apply this universally in the perp contract: if a method does not modify the pfc then it does not need to update regular fees